ui/src/Migrations/Settings.cs

```diff
@@ -0,0 +1,132 @@
+        /// </summary>
+        public void MigrateConfigAddressToSettingsFile()
+        {
+            var filename = ProductConstants.SettingsFile;
+            bool overwriteFile = false;
+
+            if (!File.Exists(filename))
+            {
+                return;
+            }
+
+            var parser = new IniParser.FileIniDataParser();
+            var data = parser.ReadFile(filename);
+
+            if (data.Sections.ContainsSection("Network"))
+            {
+                var networkData = data.Sections.GetSectionData("Network");
+                string ipAddress;
+
+                if (!networkData.Keys.ContainsKey("IPv4Address") || data["Network"]["IPv4Address"] == string.Empty)
+                {
+                    ipAddress = GetAddressFromConfig("IPv4");
+
+                    if (ipAddress != string.Empty)
+                    {
+                        data["Network"]["IPv4Address"] = ipAddress;
+                        overwriteFile = true;
+                    }
+                }
+
+                if (!networkData.Keys.ContainsKey("IPv6Address") || data["Network"]["IPv6Address"] == string.Empty)
+                {
+                    ipAddress = GetAddressFromConfig("IPv6");
+
+                    if (ipAddress != string.Empty)
+                    {
+                        data["Network"]["IPv6Address"] = ipAddress;
+                        overwriteFile = true;
+                    }
+                }
+
+                if (overwriteFile)
+                {
+                    try
+                    {
+                        FileInfo file = new FileInfo(filename);
+                        Directory.CreateDirectory(file.Directory.FullName);
+                        File.WriteAllText(filename, data.ToString());
+                    }
+                    catch (Exception e)
+                    {
+                        ErrorHandling.ErrorHandler.Handle(e, ErrorHandling.LogLevel.Error);
+                        return;
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        /// Gets IP address from config file.
+        /// </summary>
+        /// <param name="version">IP version address to be returned (IPv4 or IPv6).</param>
+        /// <returns>Address of requested IP version.</returns>
+        public string GetAddressFromConfig(string version)
+        {
+            string filePath = ProductConstants.FirefoxPrivateNetworkConfFile;
+
+            if (File.Exists(filePath))
+            {
+                var parser = new IniParser.FileIniDataParser();
+                var iniData = parser.ReadFile(filePath);
+
+                string address = iniData["Interface"]["Address"];
+                string[] addresses = address.Split(',');
+
+                IPAddress ipAddress;
+
+                foreach (var addr in addresses)
+                {
+                    if (addr != string.Empty)
+                    {
```

Comment:
Do a `continue;` here with `if (addr == string.Empty)` instead to avoid more nesting.